### PR TITLE
feat: helper for submitting fast exits

### PR DESCRIPTION
**Pull request by troggy**
Part of https://github.com/leapdao/burner-wallet/issues/4

TODO:
- [x] make it work without Metamask/Personal web3

### CODE REVIEW

**Comment by sunify** [comment]
Rewrote everything with promises. Async/await required regeneratorRuntime which is kinda big and required using transpilers

**Comment by troggy** [comment]
@sunify thanks!

However, babel-ified version of async/await code (which we publish on npm) works just okay without a runtime (in burner wallet) 🤔

File: lib/exit.js [hunk 1]
```diff
@@ -69,4 +77,73 @@ export default class Exit {
...
+      const inputs = [new Input(utxo.outpoint)];
+      const outputs = [
+        new Output(utxo.output.value, nodeConfig.exitHandlerAddr, utxo.output.color)
+      ];
+      const exitingUtxoTransfer = Tx.transfer(inputs, outputs);
+      return Exit.fastSell(exitingUtxoTransfer, plasmaChain, rootChain, marketMakerUrl, signer);
+    })
+  }
+
+  static fastSellAmount(account, amount, color, plasmaChain, rootChain, marketMakerUrl, signer) {
```
Comment (troggy) [inline lib/exit.js]: usage example for `signer` param: https://github.com/leapdao/burner-wallet/pull/24/files#diff-b00b060794e8d5797bffd335bd17e2a1R1435

Comment (sunify) [reply lib/exit.js]: Btw, this param is missing in d.ts file

**Comment by sunify** [comment]
Burner wallet uses create-react-app which uses babel (with async/await support as i remember)

File: lib/exit.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ export default class Exit {
         plasmaChain.eth.getTransaction(
           util.bufferToHex(tx.inputs[0].prevout.hash),
           signer
-          ? Promise.resolve(signer.signMessage(sigHash))
+          ? signer.signMessage(sigHash)
```
Comment (troggy) [inline lib/exit.js]: change Signer interface to return Promises

https://github.com/leapdao/burner-wallet/pull/24/commits/4c84931b73baf967797aae1016cc544279569f23

File: index.d.ts [hunk 3]
```diff
@@ -289,6 +291,16 @@ declare module "leap-core" {
     tendermintAddress: string;
   }
 
+  export type Signer = {
+    signTx: (tx: Transaction) => Promise<Tx<any>>;
```
Comment (sunify) [inline index.d.ts]: Leap transaction in parameter? In that case, you can define it like this:
```
signTx: <TxType extends Type>(tx: Tx<TxType>) => Promise<Tx<TxType>>;
```

Comment (troggy) [reply index.d.ts]: good catch! But why `<TxType extends Type>` is needed?

**Comment by troggy** [comment]
rebased on latest master

**Pull request merged**